battle-engine.js

```diff
@@ -1062,6 +1062,9 @@ class BattlePokemon {
 			if (oldAbility in {multitype:1, stancechange:1}) return false;
 		}
 		this.battle.singleEvent('End', this.battle.getAbility(oldAbility), this.abilityData, this, source, effect);
+		if (this.battle.effect && this.battle.effect.id in {entrainment:1, roleplay:1, simplebeam:1, worryseed:1}) {
```

Comment:
I don't think hardcoding is the best idea, but I don't exactly have a better idea either. :(